dotCMS/src/main/java/com/dotcms/storage/FileSystemStorage.java

```diff
@@ -0,0 +1,260 @@
+    }
+
+    @Override
+    public Object pushFile(final String bucketName,
+                       final String path,
+                       final File file,
+                       final Map<String, Object> extraMeta) {
+
+        if (!this.existsBucket(bucketName)) {
+
+            throw new IllegalArgumentException("The bucketName: " + bucketName +
+                    ", does not have any file mapped");
+        }
+
+        final File bucketFile = this.buckets.get(bucketName);
+
+        if (null != file && file.exists() && file.canRead() && bucketFile.canWrite()) {
+
+            try {
+
+                final File destBucketFile = new File(bucketFile, path);
+                FileUtils.copyFile(file, destBucketFile);
+            } catch (IOException e) {
+
+                Logger.error(this, e.getMessage(), e);
+                throw new DotRuntimeException(e);
+            }
+        } else {
+
+            throw new IllegalArgumentException("The file: " + file +
+                    ", is null, not exists or can not read. Also the bucket: " + bucketName +
+                    " could not write");
+        }
+
+        return true;
+    }
+
+    @Override
+    public Object pushObject(final String bucketName, final String path, final ObjectWriterDelegate writerDelegate,
+                             final Object object, final Map<String, Object> extraMeta) {
+
+
+        if (!this.existsBucket(bucketName)) {
+
+            throw new IllegalArgumentException("The bucketName: " + bucketName +
+                    ", does not have any file mapped");
+        }
+
+        final File bucketFile = this.buckets.get(bucketName);
+
+        if (bucketFile.canWrite()) {
+
+            try {
+
+                final File destBucketFile = new File(bucketFile, path);
+                final String compressor   = Config.getStringProperty("CONTENT_METADATA_COMPRESSOR", "none");
+                this.prepareParent(destBucketFile);
+
+                try (OutputStream outputStream = FileUtil.createOutputStream(destBucketFile.toPath(), compressor)) {
+
+                    writerDelegate.write(outputStream, object);
+                    outputStream.flush();
+                }
+            } catch (IOException e) {
+
+                Logger.error(this, e.getMessage(), e);
+                throw new DotRuntimeException(e);
+            }
+        } else {
+
+            throw new IllegalArgumentException("The bucket: " + bucketName + " could not write");
+        }
+
+        return true;
+    }
+
+    private void prepareParent(final File file) {
+
+        if (!file.getParentFile().exists()) {
+
+            file.getParentFile().mkdirs();
+        }
+    }
+
+    @Override
+    public Future<Object> pushFileAsync(final String bucketName, final String path,
+                                        final File file, final Map<String, Object> extraMeta) {
+        return DotConcurrentFactory.getInstance().getSubmitter("StoragePool").submit(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "StoragePool" appears 4 times in this file; the first occurrence is on line 160](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)